### ISSUE TITLE
Empty table headers are flagged as a failure of WCAG 1.3.1

### ISSUE THREAD

**Issue by fstrr** [dequelabs/axe-core#3404]
Product: axe Extension

Expectation:
I don't think an empty table header (e.g. `<th></th>`) is a failure of WCAG 1.3.1 Info And Relationships, which is what Axe flags and tags this code as. It's definitely best practice to have text in a table heading, but I don't think this is a WCAG failure. 1.3.1 says:

>"Information, structure, and relationships conveyed through presentation can be programmatically determined or are available in text."

If a header cell is marked up as a header cell, is in the same same row as all the other header cell, and styled differently from non-header cells, I'd argue that the presentational information, structure, and relationships are programmatically available. There's no text, so there's nothing to present visually or programmatically.

I'd expect to see this flagged as Best Practice rather than a WCAG conformance issue.

Actual:
This example is flagged in Axe with an issue tag of "wcag131":

```html
<table>
  <tr>
   <th>First Name</th>
   <th>Last Name</th>
   <th>Email Address</th>
   <th></th>
  </tr>
  <tr>
    <td>Sarah</td>
    <td>Smith</td>
    <td>sarah.smith@example.com</td>
    <td><button type="button">Edit</button><button type="button">Delete</button></td>
 </tr>
</table>
```
Motivation:
I don't believe that this is a WCAG failure, so would like to see this moved to a best practice recommendation.

<pre><code>
axe-core version: 4.4.1
Browser: Chrome 98
</code></pre>

**Comment by straker** [comment]
Thanks for the issue. So the reason we flag empty table headers as a failure of WCAG 1.3.1 is because the ARIA spec requires `columnheader` and `rowheader` [to have an accessible name](https://www.w3.org/TR/wai-aria-1.2/#columnheader). Because it's required, it can't be a best practice.

**Comment by straker** [comment]
In this particular case, to fix the issue you could replace the empty `th` with an empty `td` since there is no header for the column.

**Comment by straker** [comment]
So when mapping HTML elements to ARIA roles, the [ARIA in HTML spec](https://w3c.github.io/html-aria/#el-th) is used to determine the implicit AIRA role of the element.

As for the criterion, I think @WilcoFiers will have to answer that one.

**Comment by fstrr** [comment]
Thanks. Just to muddy the waters: Axe currently has empty headings (`h1`–`h6`) as best practice with a low user-impact [Headings must not be empty](https://dequeuniversity.com/rules/axe/3.5/empty-heading). If I look at the [ARIA in HTML spec heading role](https://w3c.github.io/html-aria/#index-aria-heading) and then the [ARIA 1.2 heading role](https://www.w3.org/TR/wai-aria-1.2/#heading), that also shows that an accessible name is required. Why are empty `h` elements a low-impact best practice but empty `th` elements a failure?

**Comment by WilcoFiers** [comment]
Makes sense I suppose. Francis would you be up for opening a pull request? You can change the tags by editing this file, and then remove the `reviewOnFail` line.
https://github.com/dequelabs/axe-core/blob/develop/lib/rules/empty-table-header.json

The impact is already set to minor so that won't need to change.

**Comment by padmavemulapati** [comment]
Validated with the latest axe-core develop branch code base with a source of `axe.js`, now `empty-table-header` rule is now best-practice  failure not with wcag incomplete issue
![image](https://user-images.githubusercontent.com/47409579/170069255-62c7f34e-9714-41a9-bfbc-ee6e8e9cff2b.png)
